c7n/mu.py

```diff
@@ -1186,8 +1192,12 @@ def add(self, func):
             log.debug("Updating cwe rule for %s" % func.event_name)
             response = self.client.put_rule(**params)
         elif not rule:
-            log.debug("Creating cwe rule for %s" % (self))
+            log.debug(f'Creating cwe rule for {self}')
             response = self.client.put_rule(**params)
+            # check if schedule exists for func.event_name and remove it to clean up
+            # when switching between scheduled event types
+            # must check all groups
+            EventBridgeScheduleSource.remove_from_groups(self.session.client('scheduler'), func)
```

Comment:
since this is new behavior, we can avoid the api call if we store state on the lambda via tag to avoid the extra api calls. that will also help clean up some of the test data set being modified.

Reply:
So to clarify, you're suggesting having a tag on the policy lambda fn to store the fact that a schedule is used for periodic mode instead of a rule? Rule based periodic mode and all other kinds of modes would not have the tag (unchanged from current), so we'd just be check for tag presence and adding/removing a tag when moving from one kind of periodic mode to the other. Am I understanding your suggestion correctly?

Reply:
Perhaps another approach would be to leave periodic mode untouched and have a new periodic-schedule mode.

Reply:
a new mode would be useful to avoid extraneous api calls on the migration, I think you'll still want to use tag state to avoid the scheduling group iteration. I think for migration purposes we could document using a custodian policy to do transition cleanup, albeit we'll also need support for scheduling groups and schedules as resources for that.

Reply:
So a new mode (`periodic-schedule`) to avoid all the extra work modifying and testing periodic to support 2 methods of implementation depending on what properties are provided in the policy, and either an additional custodian tag on the policy lambda to record the schedule group name (even if default), or perhaps another item in the custodian-info tag. I'm more concerned about the risk of breaking other things that depend on `custodian-info` containing only 2 `:` separated properties. I don't have enough visibility to know if appending to `custodian-info` is a bad idea. @kapilt What do you think?

As for migration, if you are converting an existing policy from `periodic` to `periodic-schedule`, mu.py would remove the cwe rule if I'm reading the code correctly. Shouldn't be a problem. Schedule groups as resources would be useful for tagging enforcement though. Another project for another day :-)!

Reply:
for migration, i think we would do a followup pr to support additional resources on schedule/schedule groups, and then update docs with guidance on some sample policies that can do the clean up. I'm not clear the periodic to periodic-schedule would get the cwe removal.

Reply:
Added a little to mugc.py so it can remove a schedule when one exists. This required adding an auto tag (`custodian-schedule`) to hold the group name since mugc works with the function, not the policy. Added a note to the lambda.rst doc pointing out commenting out a policy, running mugc to remove all the bits, then removing the comments and running `custodian run` again to redeploy is a simple process when switching.